ts/parts/Utilities.ts

```diff
@@ -1382,6 +1382,26 @@ function clamp(value: number, min: number, max: number): number {
     return value > min ? value < max ? value : max : min;
 }
 
+/**
+ * Utility function checking if two rectangles are overlapping
+ *
+ * @private
+ * @param {Highcharts.BBoxObject} box1 BBox of the first rectangle
+ * @param {Highcharts.BBoxObject} box2 BBox of the second rectangle
+ * @return {number} Returns a boolean value if two rectangles intersect.
+*/
+function isIntersectRect(
```

Comment:
Since we're using this function only in `overlapping-datalabels`, we can move it back to that file. If, in the future, we want to use it in other places, we can move it to Utilities and expose it.